packages/xfsprogs/0001-mkfs-source-defaults-from-config-file.patch

```diff
@@ -0,0 +1,39 @@
+From 00c4862699ef72880089cae61643728fb88d2703 Mon Sep 17 00:00:00 2001
+From: Sparks Song <shijiao@amazon.com>
+Date: Fri, 31 Jan 2025 23:18:38 +0000
+Subject: [PATCH] mkfs: source defaults from config file to make nrext64
+ default off on multiple kernel versions
+
+---
+ mkfs/xfs_mkfs.c | 8 ++++++++
+ 1 file changed, 8 insertions(+)
+
+diff --git a/mkfs/xfs_mkfs.c b/mkfs/xfs_mkfs.c
+index 6d2469c..91a0f37 100644
+--- a/mkfs/xfs_mkfs.c
++++ b/mkfs/xfs_mkfs.c
+@@ -39,6 +39,9 @@
+  */
+ #define WHACK_SIZE (128 * 1024)
+
++/* Default path for the mkfs.xfs configuration file */
++#define DEFAULT_CONFIG_PATH "/usr/share/xfs/mkfs.xfs.conf"
++
+ /*
+  * XXX: The configured block and sector sizes are defined as global variables so
+  * that they don't need to be passed to getnum/cvtnum().
+@@ -4428,6 +4431,11 @@ main(
+ 	 * the options from this file parsed, we can then proceed with parameter
+ 	 * and bounds checking and making the filesystem.
+ 	 */
++	if (cli.cfgfile == NULL) {
++		//check if default config file is there
```

Comment:
You will need to add an actual check that the file exists.

If it doesn't exist, then this would result in a fatal error until `mkfs.xfs` was invoked with config file that did exist.

Since you don't have any package-level dependencies that would ensure the config file is present, you must be defensive and assume it won't be there.